packages/crawler/src/evaluate-page.ts

```diff
@@ -0,0 +1,155 @@
+import * as puppeteer from 'puppeteer';
+import { TagName } from 'shared/src';
+import { ConsumedTagNamesForVersionsAndPrefixes, TagNameData, TagNamesWithPropertyNames } from './types';
+
+declare global {
+  // eslint-disable-next-line @typescript-eslint/consistent-type-definitions
+  interface Document {
+    // Extend Document interface, so we don't have to cast it on any
+    porscheDesignSystem: { [version: string]: { prefixes: string[] } };
+  }
+}
+
+export const evaluatePage = async (
+  page: puppeteer.Page,
+  pdsTagNamesWithPropertyNames: TagNamesWithPropertyNames
+): Promise<ConsumedTagNamesForVersionsAndPrefixes> => {
+  const pdsCrawlerRawData = await page.evaluate(
+    async ({ pdsTagNamesWithPropertyNames }): Promise<ConsumedTagNamesForVersionsAndPrefixes> => {
+      const tagNames = Object.keys(pdsTagNamesWithPropertyNames);
+
+      const getPdsComponentNameByPrefix = (el: Element, prefix: string): TagName | null => {
+        const tagName = el.tagName.toLowerCase();
+        return (
+          (Object.keys(pdsTagNamesWithPropertyNames).find(
+            (compName) => (prefix ? `${prefix}-${compName}` : compName) === tagName
+          ) as TagName) || null
+        );
+      };
+
+      const getAllChildElements = (el: Element): Element[] => {
+        const children = Array.from(el.children)
+          .concat(Array.from(el.shadowRoot?.children || []))
+          .flat() as Element[];
+        return children.concat(children.map(getAllChildElements).flat());
+      };
+
+      const getChildren = (el: Element): string | null =>
```

Comment:
Function name doesn't really match what's happening?

Reply:
true, renamed the function